source/type/type.h

```diff
@@ -520,6 +522,22 @@ struct Contributor : public Header, Nameable{
     bool operator<(const Contributor & other) const { return this < &other; }
 };
 
+struct Frame : public Header{
+    const static Type_e type = Type_e::Frame;
+    Contributor* contributor=nullptr;
+    navitia::type::RTLevel realtime_level = navitia::type::RTLevel::Base;
+    boost::gregorian::date_period validation_date;
+    std::string desc;
+    std::string system;
+
+    Frame(): validation_date(boost::gregorian::date(), boost::gregorian::date()){}
+
+    template<class Archive> void serialize(Archive & ar, const unsigned int ) {
+        ar & idx & uri & contributor & realtime_level & validation_date & desc & system;
+    }
+    bool operator<(const Frame & other) const { return this < &other; }
```

Comment:
same as the other class:
comparing the idx is better I think, some minor indent problems, and the constructor for validation_date

btw wouldn't it be better to call it validation_period ?